File: python/polyreg.py
import pandas as pd

# Import dataset
df = pdf.read_csv('data.csv')
X = df.iloc[:, :-1].values
y = df.iloc[:, -1].values

# Transform features to polynomials
from sklearn.preprocessing import PolynomialFeatures
poly_features = PolynomialFeatures(degree=4)
X_poly = poly_features.fit_transform(X)

# Split into training and test sets
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X_poly, y, test_size=0.2)

# Fit polynomial linear regression to training set
poly_model = LinearRegression()
poly_model.fit(X_train, y_train)

# Predict test set
y_pred = model.predict(X_test)

File: r/polyreg.r
# Import dataset
df = read.csv('data.csv')

# Create Boolean vector with random split
library(caTools)
split = sample.split(df$y, SplitRatio=0.8)

# Split into training and test set
train = subset(df, split == TRUE)
test = subset(df, split == FALSE)

# Fit polynomial linear regression to training set
poly_model = lm(y ~ polym(x1, x2, x3, degree=4), data=train)

# Predict test set
y_pred = predict(poly_model, newdata=test)